runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java

```diff
@@ -1708,4 +1709,73 @@ public SslCertificate getCertificate() {
         checkThreadSafety();
         return mContent.getCertificate();
     }
+
+    /**
+     * Registers the listener to be notified as find-on-page operations progress.
+     *
+     * @param listener an implementation of {@link XWalkFindListener}
+     * @since 7.0
+     */
+    @XWalkAPI(reservable = true)
+    public void setFindListener(XWalkFindListenerInternal listener) {
+        if (mContent == null) return;
+        checkThreadSafety();
+        mContent.setFindListener(listener);
+    }
+
+    /**
+     * Finds all instances of find on the page and highlights them asynchronously.
+     * Notifies any registered {@link XWalkFindListener}.
+     * Successive calls to this will cancel any pending searches.
+     *
+     * @param searchString the string to find.
+     * @since 7.0
+     */
+    @XWalkAPI
+    public void findAllAsync(final String searchString) {
+        if (mContent == null) return;
+        ThreadUtils.runOnUiThread(new Runnable() {
```

Comment:
checkThreadSafety ensures that this can only be called in main thread(UI thread).   With ThreadUtils.runOnUiThread, this can be call in any thread, although the real task is done in UI thread.

My only concern for this is,  can user call this after load soon?  Should we post this in a loop till it is ready for search?